### PR TITLE
feat(evmutil): add ERC20KavaWrappedNativeCoinContract

### PR DESCRIPTION
**Pull request by pirtleshell**
## Description
* adds the contract ABI & bytecode for an Ownable ERC20 with the following:
  * customizable decimals on deploy -> requires overriding decimals() view
  * mint() exposed for the contract owner which will be the evmutil module
  * burn() exposed for the contract owner which will be the evmutil module
* sets up keeper to deploy above token based on details from an AllowedNativeCoinERC20Token
* tests basic queries and permissions of deployed contract

The chain does not use this new deploy method yet. I intend to add `MsgConvertNativeCoinToERC20` in my next PR. It will use this deployment code upon first conversion of tokens defined in the `allowed_native_denoms` param.

See contract code here: #1594 (this branch is based on that pr)

## Checklist
 - [x] Changelog has been updated as necessary.

### CODE REVIEW

**Comment by pirtleshell** [comment]
converted to draft until merge of #1594 which pulls the contract directly into this repo.

will rebase onto that once reviewed & merged.

File: x/evmutil/keeper/erc20.go
Comment (galxy25) [inline x/evmutil/keeper/erc20.go:67]: missing public method comment

File: x/evmutil/keeper/erc20.go
Comment (galxy25) [inline x/evmutil/keeper/erc20.go:72]: might be useful to log what token denom failed

File: x/evmutil/keeper/erc20.go
Comment (galxy25) [inline x/evmutil/keeper/erc20.go:103]: also useful to log the data and nonce here

File: x/evmutil/types/contract.go
Comment (galxy25) [inline x/evmutil/types/contract.go:57]: might be useful to log the json in the panic as well

**Review by galxy25 — APPROVED** [review]
approved pending resolutions of comments

**Pull request merged**
